### PR TITLE
Enable prettier to format files

### PR DESCRIPTION
**Pull request by korthout**
Here's a first setup to format files using prettier. I recommend reading the commits individually, as I've described the intensions in more detail there.

Generally:
- You can run it using `npm run format`. 
- By removing entries from the `.prettierignore` file, we can transition to formatting the files in steps.
- A few first files have been formatted to show case how to continue.
- CI-integration to verify that everything is formatted, still needs to be added. For this we could use `prettier --check .`

See #299

### CODE REVIEW

**Comment by korthout** [comment]
I've rebased on top op `master` to resolve the conflicts.

**Comment by akeller** [comment]
@pepopowitz FYI :)

**Comment by pepopowitz** [comment]
This is amazing, @korthout! Thanks for doing this. 

1. To confirm that I understand your expectations with this PR: 

- In-scope of this PR, preventing us from merging

  - set up a prettier github action to fail when files are introduced without being formatted

- Out-of-scope of this PR, and coming in follow-up PRs

  - uncomment the remaining folders in .prettierignore, and format the associated files. This can be done one folder at a time.

2. If I've got that accurate, are you intending on working on setting up the github action soon, or is this something you'd prefer I pick up? (I love to pair so it's also something we could do together 🙂 )

3. What do you think about the idea of adding a [git pre-commit hook](https://prettier.io/docs/en/precommit.html) that runs the formatter, to minimize the chances of someone pushing up an unformatted commit, seeing the github action fail, and pushing up an ensuing "format" commit?

**Comment by korthout** [comment]
Hi @pepopowitz, sorry for the late reply. I'm actually out sick, but still want to quickly answer your questions.

> * In-scope of this PR, preventing us from merging
>   
>   * set up a prettier github action to fail when files are introduced without being formatted

IMO, this is something we need to have rather sooner than later. Otherwise, there is no guarantee that formatting is applied. This would eventually likely lead to someone having to run the formatter once in a while, which in turn might then lead to changes in many files which is prone to merge conflicts. In these cases, it's generally best just to verify it in the CI. 

Another option is to auto-apply it in the CI and allow a bot to make and push commits. Personally, I'm not a fan of this because unreviewed changes could lead to unforeseen mistakes. I'd argue for verification in CI over application in CI.

I don't necessarily think it must be part of this PR. It's fine to tackle this shortly after this PR.

> * Out-of-scope of this PR, and coming in follow-up PRs
>   
>   * uncomment the remaining folders in .prettierignore, and format the associated files. This can be done one folder at a time.

Absolutely. I'd argue that the docs folder doesn't have to be done all at once, but its entry in the .prettierignore can be replaced by the files/folders it contains which can then be done one folder at a time. Even though the formatting is applied automatically, I believe someone should review the changes and PRs shouldn't be too big.

> 2. If I've got that accurate, are you intending on working on setting up the github action soon, or is this something you'd prefer I pick up? (I love to pair so it's also something we could do together 🙂 )

Originally, I wanted to just get it all working 😆 But, I just didn't have the time to finish it completely.

If you want, I'd be happy to hand over the topic to you. Perhaps you can review this PR and then continue with the CI stuff. Of course, I can also stay involved if you want me to. Would you be willing to take it over?

> 3. What do you think about the idea of adding a [git pre-commit hook](https://prettier.io/docs/en/precommit.html) that runs the formatter, to minimize the chances of someone pushing up an unformatted commit, seeing the github action fail, and pushing up an ensuing "format" commit?

I'm not so familiar with using pre-commit hooks, TBH. In the Zeebe team, we generally build the project locally with maven which also auto-applies the formatting. For documentation, that might be different because writing docs and building the project might be unrelated for some devs. So, it might make more sense to run a pre-commit hook. I think we can just try this. If devs complain because it breaks their git operations or it takes too much time, it can always be reverted. WDYT?

**Comment by pepopowitz** [comment]
Thanks for responding, Nico, hope you feel better soon. It looks like we're on the same page.

> Another option is to auto-apply it in the CI and allow a bot to make and push commits. Personally, I'm not a fan of this because unreviewed changes could lead to unforeseen mistakes. I'd argue for verification in CI over application in CI.

+1, I have the same opinion.

> ...the docs folder doesn't have to be done all at once, but its entry in the .prettierignore can be replaced by the files/folders it contains which can then be done one folder at a time. Even though the formatting is applied automatically, I believe someone should review the changes and PRs shouldn't be too big.

+1 to small PRs 

> Would you be willing to take it over?

I think so! I'll be meeting later today with @akeller and I'll talk to her about where this could fit in with my priorities, and I'll let you know where we settle afterward. 

> [re: git hooks] I think we can just try this. If devs complain because it breaks their git operations or it takes too much time, it can always be reverted. WDYT?

Yes, this sounds great. 

---

Given all that, here's a list that I think captures our intentions -- let me know if anything's missing or I misunderstood.

- [ ] This PR: ready for review
- [ ] Followup PR: Add a GitHub Action to run prettier
- [ ] Followup PRs: Include remaining folders in prettier checks
- [ ] Followup PR: Add a pre-commit hook to auto-format files when they're updated

File: .github/workflows/publish-stage.yaml
Comment (pepopowitz) [inline .github/workflows/publish-stage.yaml:23]: (no action required) 

This looks right to me, but just leaving a placeholder for myself to remember that this part of the staging deploy changed, in case it ends up being an issue.

File: sidebars.js
Comment (pepopowitz) [inline sidebars.js:111]: ❤️ yesssss, trailing commas

File: sidebars.js
Comment (pepopowitz) [inline sidebars.js:408]: Potentially blocking: 

There's one removed item here (`kubernetes-helm`) -- is that intentional?

File: sidebars.js
Comment (pepopowitz) [inline sidebars.js:441]: Potentially blocking:

Also several removed here. Are these old, and that's why they were removed?

**Review by pepopowitz — COMMENTED** [review]
This looks great! 

There were a few lines removed that I wasn't sure about -- I want to confirm they were intentional removals before approving/merging.

**Comment by pepopowitz** [comment]
Following up re: taking over this work -- @akeller and I talked about it today and yes, I will own this moving forward!

File: .github/workflows/publish-stage.yaml
Comment (korthout) [reply .github/workflows/publish-stage.yaml:23]: Good idea 👍 

TBH, this one was a surprise to me too, but apparently, YAML escapes quotes `'` with another quote `''`.

File: sidebars.js
Comment (korthout) [reply sidebars.js:111]: 👍 Of course, this is just the default setting. I think it's good, but if we prefer something else we can change it in the `.prettierrc.json`

File: sidebars.js
Comment (korthout) [reply sidebars.js:408]: Nope, good catch. Might have happened in the conflict resolution

File: sidebars.js
Comment (korthout) [reply sidebars.js:441]: Also unintentional. Good catch

**Review by pepopowitz — APPROVED** [review]
Sorry for the delayed response @korthout! This looks great, and I'll plan to either merge tomorrow, or Monday. I will do my best to take care of any merge conflicts that show up by then, but might reach out for help if any tricky ones appear.

**Pull request merged**
